### PR TITLE
active_shipping 1.x

### PR DESCRIPTION
**Pull request by wvanbergen**
Prep for major release:

- Changes namespace from `ActiveMerchant::Shipping` to `ActiveShipping`
- Get rid of `ActiveShipping::Base`. `AS::Base.mode` is gone, and `AS::Base.carrier` now is `ActiveShipping::Carrier.find`
- Use autoloading to load carriers, so we don't load all if them if only one is needed.
- Bump active_utils dependency to 3.x
- Bump activesupport dependency to >= 3.2
- Run CI against multiple activesupport versions

@RichardBlair @pickle27 @ntalbott

### CODE REVIEW

**Comment by wvanbergen** [comment]
Question: what is the stuff in `vendor` (quantified / xml_node).  Can we get rid of it / use real gem dependencies?

**Comment by wvanbergen** [comment]
@RichardBlair @robfoster I updated the test code; I haven't changed anything to the library itself. Can you do a quick review? The remote tests are still TODO.

**Comment by wvanbergen** [comment]
Anything else we want to tackle if we are doing a major release anyway?

**Comment by robfoster** [comment]
lgtm

**Comment by RichardBlair** [comment]
This is on my plate to review today

**Comment by wvanbergen** [comment]
@RichardBlair Looking at the diff won't be very useful due to the amount of changes.

The only changes to production code is the namespace change from `ActiveMerchant::Shipping` to `ActiveShipping`. Other than that the API stays the same. All the other changes are related to dependency updates, moving to Minitest, and CI configuration changes.

File: lib/active_shipping.rb [hunk 1]
```diff
@@ -37,17 +29,17 @@
...
-require 'active_shipping/shipping/label_response'
-require 'active_shipping/shipping/package'
-require 'active_shipping/shipping/location'
-require 'active_shipping/shipping/rate_estimate'
-require 'active_shipping/shipping/shipment_event'
-require 'active_shipping/shipping/shipment_packer'
-require 'active_shipping/shipping/carrier'
-require 'active_shipping/shipping/carriers'
-require 'active_shipping/shipping/errors'
+require 'active_shipping/base'
```
Comment (RichardBlair) [inline lib/active_shipping.rb]: :+1:

File: lib/active_shipping/shipping/carrier.rb [hunk 2]
```diff
@@ -1,84 +0,0 @@
-module ActiveMerchant
```
Comment (RichardBlair) [inline lib/active_shipping/shipping/carrier.rb]: :heart:

**Comment by RichardBlair** [comment]
The namespaces of active shipping has bothered me for a long time, and has been on my backlog. Thanks so much for taking care of this.

The diff looks fine. Once CI passes and we verify all is well in the force, we can definitely :ship: this.

**Comment by wvanbergen** [comment]
Going to merge to master this and do a pre-release. If we want to do more for this major release, we can use separate PRs.

**Pull request merged**